### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by kattni**

### CODE REVIEW

**Comment by kattni** [comment]
Tested on Hollowing with invalid file present on board.

File: adafruit_slideshow.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                yield image
+            if not self._loop:
+                return
+
+    def _load_images(self):
+        """Loads the list of images to be displayed in alphabetical or random order."""
+        self._file_list = self._get_filenames()
+        if self._order == SlideShow.RANDOM:
+            self._file_list = sorted(self._file_list, key=lambda x: random.random())
+            self._images = self._get_next_image()
```
Comment (caternuson) [inline adafruit_slideshow.py]: Unindent. Shouldn't be in the `if`. Looks like that was my bad. I only tested random. :(

File: adafruit_slideshow.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    """
+    MAX_BRIGHTNESS = 2 ** 15
+
+    ALPHA = 0
+    RANDOM = 1
+
+    FADE_IN = 0
+    SHOW_IMG = 1
+    FADE_OUT = 2
+    LOAD_IMG = 3
```
Comment (sommersoft) [inline adafruit_slideshow.py]: These are all internal variables. Should start with `_`? (`_FADE_IN`)

Also, now that I read further down...these should also move outside of the `SlideShow()` class. The variables themselves don't change, the `self._current_state` does, so you don't need to recreate them for every instance of the class. Calls to `self.FADE_IN` etc will also need to change after that.

File: adafruit_slideshow.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        self._folder = folder
+
+    @property
+    def order(self):
+        """Specifies the order in which the images are displayed. Options are random (``RANDOM``) or
+        alphabetical (``ALPHA``). Default is ``RANDOM``."""
+        return self._order
+
+    @order.setter
+    def order(self, order):
```
Comment (sommersoft) [inline adafruit_slideshow.py]: If there are only 2 options, I would check that the supplied argument meets one of them:
```python
def order(self, order):
    if not order in [ALPHA, RANDOM]:
        raise ValueError("Order must be either 'RANDOM' or 'ALPHA'")
```
This should also be checked in `__init__`.

File: adafruit_slideshow.py (hunk 1 above)
Comment (kattni) [reply adafruit_slideshow.py]: I was testing it with alphabetical and it seemed to be working so I didn't think anything of it. On closer investigation it was only mostly working. Fixed.

File: adafruit_slideshow.py (hunk 3 above)
Comment (kattni) [reply adafruit_slideshow.py]: So every instance of `_order` gets changed to `order` then?

**Comment by jerryneedell** [comment]
I found a minor bug in the slideshow_simpletest.py:
```
change
slideshow.order = slideshow.ALPHA
to
slideshow.order = adafruit_slideshow.PlayBackMode.ALPHA
```

**Comment by kattni** [comment]
@jerryneedell I already found that and updated it. Thanks for catching it though! Travis failed on it for me.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+.. code-block:: python
+
+    import adafruit_slideshow
+
+    slideshow = adafruit_slideshow.SlideShow()
+    slideshow.loop = False
+    slideshow.order = slideshow.ALPHA
+
+    while slideshow.update():
+        pass
```
Comment (caternuson) [inline README.rst]: `ALPHA` is in the `PlayBackMode` class now. Check the rest of the example as well.

File: adafruit_slideshow.py [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        import adafruit_slideshow
+
+        slideshow = adafruit_slideshow.SlideShow()
+        slideshow.loop = False
+        slideshow.order = PlayBackMode.ALPHA
+
+        while slideshow.update():
+            pass
+
+    Example code for Hollowing Express. Sets ``dwell`` to 0 seconds, turns ``auto_advance`` off,
```
Comment (caternuson) [inline adafruit_slideshow.py]: s/Hollowing/Hallowing

File: adafruit_slideshow.py [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+display.
+
+* Author(s): Kattni Rembor, Carter Nelson, Roy Hooper
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+ * `Adafruit Hollowing M0 Express <https://www.adafruit.com/product/3900>`_
```
Comment (caternuson) [inline adafruit_slideshow.py]: s/Hollowing/Hallowing

File: adafruit_slideshow.py [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+    :param fade_effect: Specify whether to include the fade effect between images. ``True``
+                        tells the code to fade the backlight up and down between image display
+                        transitions. ``False`` maintains max brightness on the backlight between
+                        image transitions. Default is ``True``.
+
+    :param auto_advance: Specify whether to automatically advance after dwell seconds. ``True``
+                 if slideshow should auto play, ``False`` if you want to control advancement
+                 manually.  Default is ``True``.
+
+    Example code for Hollowing Express. With this example, the slideshow will play through once
```
Comment (caternuson) [inline adafruit_slideshow.py]: s/Hollowing/Hallowing

File: docs/index.rst [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    api
+
+.. toctree::
+    :caption: Tutorials
+
+
+.. toctree::
+    :caption: Related Products
+
+    Adafruit Hollowing M0 Express <https://www.adafruit.com/product/3900>
```
Comment (caternuson) [inline docs/index.rst]: s/Hollowing/Hallowing

**Comment by tannewt** [comment]
The extra states jumped out at me because each `if` for the current state wasn't exclusive (`elif`). That meant that in practice you'd transition through multiple states (_FADE_OUT always goes to _LOAD_IMAGE for example) in a single call each time, which is a good sign you don't need them separate.

**Comment by kattni** [comment]
In commit `caee24a`, added the ability to call the name of the current image being displayed with `slideshow.current_image_name`. Apparently using markdown in a commit message caused it to not include the rest of the commit message.

**Pull request merged**